docker/fetch_normalization_data.py

```diff
@@ -0,0 +1,194 @@
+    try:
+        cursor.execute("""
+            INSERT INTO product_adu (adu_count, adu_date, os_name, product_version_id)
+            VALUES (%s, %s, %s, %s)
+        """, (adu_count, adu_date, os_name, product_version_id))
+    except psycopg2.IntegrityError as exc:
+        # Hitting an IntegrityError here probably means this data is in the table already, so we
+        # just skip it.
+        pass
+    conn.commit()
+
+
+def get_connection():
+    """Builds a connection using ConnectionContext environment variables
+
+    :returns: postgres connection
+
+    """
+    # This uses the same environment variables and defaults as postgres ConnectionContext
+    host = os.environ.get('resource.postgresql.database_hostname', 'localhost')
+    port = int(os.environ.get('resource.postgresql.database_port', '5432'))
+    username = os.environ.get('secrets.postgresql.database_username', 'breakpad_rw')
+    password = os.environ.get('secrets.postgresql.database_password', 'aPassword')
+    dbname = os.environ.get('resource.postgresql.database_name', 'breakpad')
+
+    local_config = {
+        'database_hostname': host,
+        'database_port': port,
+        'database_username': username,
+        'database_password': password,
+        'database_name': dbname
+    }
+    dsn = (
+        "host=%(database_hostname)s "
+        "dbname=%(database_name)s "
+        "port=%(database_port)s "
+        "user=%(database_username)s "
+        "password=%(database_password)s"
+    ) % local_config
+    return psycopg2.connect(dsn)
+
+
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--products', default='Firefox',
+        help='comma separated set of products to fetch ADI for'
+    )
+    parser.add_argument(
+        '--start_date', default='LASTWEEK', help='start date (yyyy-mm-dd)'
+    )
+    parser.add_argument(
+        '--end_date', default='TODAY', help='end date (yyyy-mm-dd)'
+    )
+    parser.add_argument(
+        '--endpoint', default='https://crash-stats.mozilla.com/api/ADI/',
+        help='endpoint for ADI API'
+    )
+    args = parser.parse_args()
+
+    today = datetime.datetime.utcnow()
+    last_week = today - datetime.timedelta(days=7)
+
+    start_date = args.start_date
+    if start_date == 'LASTWEEK':
+        start_date = last_week.strftime('%Y-%m-%d')
+
+    end_date = args.end_date
+    if end_date == 'TODAY':
+        end_date = today.strftime('%Y-%m-%d')
+
+    if start_date > end_date:
+        print('Start date (%s) cannot be after end date (%s). Exiting.' % (start_date, end_date))
+        return 1
+
+    conn = get_connection()
+
+    print('Fetching normalization data for %s from %s to %s' % (
+        args.products, start_date, end_date)
+    )
+
+    for product in args.products.split(','):
+        # Fetch versions from the product_versions table that we should pull data for
+        versions = get_versions(conn, product)
+
+        for platform in PLATFORMS:
```

Comment:
Nit: If there's a reason for not fetching for all platforms all at once perhaps a comment here. 
After all, it *could* be this:
```python
        print('Fetching data for (%s, %s, %s)...' % (product, versions, PLATFORMS))
        adu_data = fetch_data_from_api(
            endpoint=args.endpoint,
            product=product,
            platform=PLATFORMS,
            versions=versions,
            start_date=start_date,
            end_date=end_date
        )
```
But consider it a nit. It's not painful to have to do 4 API calls instead of 1.